glean-core/python/tests/test_dispatcher.py

```diff
@@ -18,13 +18,13 @@ def update():
     for i in range(3):
```

Comment:
Since we're adding threading to the mix, we should probably port over some of the related tests from [DispatchersTest.kt](https://searchfox.org/glean/source/glean-core/android/src/test/java/mozilla/telemetry/glean/DispatchersTest.kt). I'm specifically thinking off:

- verify that tasks are dispatched on a separate thread when we're not in test mode;
- verify that, when running off the main thread, tasks are executed in the order they are received;
- exceptions thrown in the dispatched tasks do not break glean.